### ISSUE TITLE
API connector: TypeError [ERR_UNESCAPED_CHARACTERS]

### ISSUE THREAD

**Issue by netwarex** [chartbrew/chartbrew#175]
Hi,

I found out a bug, if the API connector contains "-" or any other escapable characters it makes the request fail:

```
{
  "statusCode": 400,
  "statusText": "Bad Request",
  "message": "{\"name\":\"RequestError\",\"message\":\"TypeError [ERR_UNESCAPED_CHARACTERS]: Request path contains unescaped characters\",\"cause\":{\"code\":\"ERR_UNESCAPED_CHARACTERS\"},\"error\":{\"code\":\"ERR_UNESCAPED_CHARACTERS\"},\"options\":{\"url\":\"https://api.transferwise.com/v1/profiles/12345678/balance‑statements/12345678/statement.json\",\"method\":\"GET\",\"headers\":{\"Authorization\":\"Bearer xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx\"},\"qs\":{},\"resolveWithFullResponse\":true,\"simple\":false,\"transform2xxOnly\":false}}"
}
```

Here the part is "balance‑statements" which makes this error.

**Comment by razvanilin** [comment]
Good catch! I'll have a look into this and see what's up 👌

**Comment by razvanilin** [comment]
So I don't think it's the `-` because that's a valid character for URLs and I tested some of my APIs with those and it worked. Maybe there's another sneaky character entered through copy-pasting the URL in?

I just added a new commit that adds an extra step to encode the URL just in case. The new changes are in the `master` branch or if you use Docker, I just pushed a new image with the new changes. Let me know how it goes ✌️
